### PR TITLE
WIP: Reorganisation of Docker Guide

### PR DESCRIPTION
**Pull request by tomschr**
### Description

The following pull request improves the structure in some many places that it is hard to summarize it.

### Checklist 

for Martin:

* [ ] fix `<remark>`s from @taroth21 (see below)  and @tomschr; when you have fixed a remark, don't forget to remove the `<remark>` tag.
* [x] fix conflict (seems easy to fix)
* [x] integrate PR #323 and close it when done
* [ ] squash all the commits into a sensible amount :wink: when done
* [ ] mark the card in Trello as "Done" and add the link of this PR or commit as comment (or attachment)

### Relations

* PR #323 
* https://trello.com/c/VXD3N2ix/385-reorg-docker-guide

### CODE REVIEW

File: xml/docker_building_images.xml [hunk 1]
```diff
@@ -8,68 +8,74 @@
...
  xmlns:xi="http://www.w3.org/2001/XInclude"
  xmlns:xlink="http://www.w3.org/1999/xlink">
- <title>Creating Custom Images</title>
+ <title>Creating Custom &docker; Images</title>
  <para>
-  For creating your custom image you need a base &docker; image of &slsa;. You
-  can use any of the pre-built &slsa; images that you can obtain as described
-  in <xref linkend="Customizing_Pre-build_Images"/>.
+  For creating your custom image, you need a base &docker; image of &slsa;.
+  You can use any of the pre-built &slsa; images that you can obtain as
```
Comment (taroth21) [inline xml/docker_building_images.xml]: Proposal: You can use any of the pre-built images of &productname;. Download them as decribed in ```<xref .../>```.

File: xml/docker_building_images.xml [hunk 2]
```diff
@@ -8,68 +8,74 @@
...
-  <para>
-   Pre-built images do not have repositories configured. But when the &docker;
-   host has a &slea; subscription that provides access to the product used in the
-   image, Zypper will automatically have access to the right repositories.
-  </para>
+     <procedure>
+      <title>Activating the Base Image for up to &slsa;&nbsp;12 SP2</title>
+      <step>
+       <para>
+        Get the proper image name with <literal>sle2docker</literal> by running
```
Comment (taroth21) [inline xml/docker_building_images.xml]: end the para with a colon: "by running:"

File: xml/docker_building_images.xml [hunk 3]
```diff
@@ -8,68 +8,74 @@
...
+      </step>
+      <step>
+       <para>
+        Activate the image by using the image name from the previous step:
+       </para>
+       <screen>&prompt.user;sle2docker activate <emphasis>PRE-BUILT_IMAGE_NAME</emphasis></screen>
+      </step>
+      <step>
+       <para>
+        Check if the image was successfully activated by running
```
Comment (taroth21) [inline xml/docker_building_images.xml]: you can even skip "by running". Maybe shorten to: "Check if the image was successfully activated:"

File: xml/docker_building_images.xml [hunk 4]
```diff
@@ -8,68 +8,74 @@
...
+      </step>
+     </procedure>
+    </listitem>
+   </varlistentry>
+   <varlistentry>
+    <term>&sle;&nbsp;12 SP3 and later</term>
+    <listitem>
+     <screen>&prompt.user;docker pull registry.suse.com/suse/<replaceable>IMAGENAME</replaceable></screen>
+     <para>
+      For example, for &productname;&nbsp;15, use
```
Comment (taroth21) [inline xml/docker_building_images.xml]: add missing colon at the end: "use:"

File: xml/docker_concept.xml [hunk 5]
```diff
@@ -0,0 +1,498 @@
...
+ </para>
+ <sect1 xml:id="sec.docker.installation.terminology_deng">
+  <title>&deng; Terminology</title>
+  <remark>toms 2019-02-04: check with Tanja, if you need the full title or
+  if "Terminology" is enough.</remark>
+  <para>
+   Before going into detail about &deng;, let us define some of the terms used:
+  </para>
+  <remark>toms 2019-02-04: two things:
+   1. Does it makes sense to sort the list alphabetically? It seems, we
```
Comment (taroth21) [inline xml/docker_concept.xml]: Yep, sort alphabetically, I would say...

File: xml/docker_concept.xml [hunk 6]
```diff
@@ -0,0 +1,498 @@
...
+      remove the "A registry is..." and write "Storage for already created
+      images." Same for the other terms.
+  </remark>
+
+  <variablelist>
+   <varlistentry xml:id="Docker_engine">
+    <term>&deng;</term>
+    <listitem>
+     <para>
+      &deng; is a server-client type application that performs all
```
Comment (taroth21) [inline xml/docker_concept.xml]: Shorten to: "A server-client type..."

File: xml/docker_concept.xml [hunk 7]
```diff
@@ -0,0 +1,498 @@
...
+         The server side of &deng; manages all &docker; objects (images,
+         containers, network connections used by containers, etc.).
+        </para>
+       </formalpara>
+      </listitem>
+      <listitem>
+       <formalpara>
+        <title>REST API:</title>
+        <para>
+         Applications can use this API to communicate directly with the daemon.
```
Comment (taroth21) [inline xml/docker_concept.xml]: Use a similar structure like in the other explanations. As you want to define the term (what is the REST API?), I would rephrase to:
"An API which can be used to directly communicate with the daemon."

File: xml/docker_concept.xml [hunk 8]
```diff
@@ -0,0 +1,498 @@
...
+     <para>
+      An <emphasis>image</emphasis> is a read-only template<remark>toms 2019-02-04: 
+      template sounds a bit strange in this context.
+      On Quora.com, I've found an interesting definition:
+      "An image is a compressed, self-contained piece of software which has
+      to be unwrapped in order to use the functionality." Maybe you can add
+      it somehow.</remark> used to create a
+      <emphasis>virtual machine</emphasis> on the host server. A &docker; image is
+      made by a series of layers built one over the other. Each layer
+      corresponds to a permanent change, for example an update of an
```
Comment (taroth21) [inline xml/docker_concept.xml]: not sure what you want to say with "permanent change"? in fact, it seems a bit contradictory to me -  a change is something that happens at a given point in time (or during a transition period), how can this be permanent at the same time?

File: xml/docker_concept.xml [hunk 9]
```diff
@@ -0,0 +1,498 @@
...
+         on a different machine than the CLI client, the CLI client can
+         communicate by using network sockets or the REST API provided by
+         &deng;.
+        </para>
+       </formalpara>
+      </listitem>
+     </itemizedlist>
+    </listitem>
+   </varlistentry>
+   <varlistentry xml:id="Docker_Image">
```
Comment (taroth21) [inline xml/docker_concept.xml]: Maybe add 'Docker' to the term, at least in brackets: " (Docker) Image"
('image' is a very generic term and is often used in totally different contexts)

File: xml/docker_concept.xml [hunk 10]
```diff
@@ -0,0 +1,498 @@
...
+      "An image is a compressed, self-contained piece of software which has
+      to be unwrapped in order to use the functionality." Maybe you can add
+      it somehow.</remark> used to create a
+      <emphasis>virtual machine</emphasis> on the host server. A &docker; image is
+      made by a series of layers built one over the other. Each layer
+      corresponds to a permanent change, for example an update of an
+      application.
+      <remark>toms 2019-02-04: The following sentence is dangerous and
+      misleading IMHO. The term "changes" does not relate to a Dockerfile.
+      That file is used merely as a kind of "build recipe".
```
Comment (taroth21) [inline xml/docker_concept.xml]: I agree with Toms here. Maybe rephrase to something like this:
"A self-contained piece of software. A Docker image is a read-only template used to create a virtual machine on
the host server. The image is built with a Dockerfile." (I would give the explanation for Dockerfile only in the varlistentry on 'Dockerfile').

File: xml/docker_concept.xml [hunk 11]
```diff
@@ -0,0 +1,498 @@
...
+      I would propose to rephase/move the sentence at the beginning of this
+      paragraph:
+      "An image is a read-only template used to create a virtual machine on
+      the host server. Such an image is represented/build by using a
+      Dockerfile which acts as an "recipe". A Docker image is..."
+      </remark>
+      The changes are stored in a file called a
+      <emphasis>Dockerfile</emphasis>. For more details see
```
Comment (taroth21) [inline xml/docker_concept.xml]: Because the sentence before is about the Dockerfile, I would mention that following link is about the Docker image:
"For more details on Docker images, see  [...]."

File: xml/docker_concept.xml [hunk 12]
```diff
@@ -0,0 +1,498 @@
...
+      <remark>toms 2019-02-04: The description is misleading IMHO. It gives
+       it a different spin. ;)
+      According to https://docs.docker.com/engine/reference/builder/
+      "A Dockerfile is a text document that contains all the commands a user
+      could call on the command line to assemble an image. Using docker build
+      users can create an automated build that executes several command-line
+      instructions in succession."
+      I would suggest to rephrase your explanation.
+      </remark>
+      A <emphasis>Dockerfile</emphasis> stores changes made on top of the base
```
Comment (taroth21) [inline xml/docker_concept.xml]: Following Toms' proposal, I would suggest to rephrase to something along those lines: "A text document which contains commands to assemble a Docker image. With the ```docker build``` command, the instructions provided in the Dockerfile are executed to create a Docker image. The Dockerfile can store modifcations you want to apply to the resulting image (on top of a base image)."

File: xml/docker_concept.xml [hunk 13]
```diff
@@ -0,0 +1,498 @@
...
+      <emphasis>Dockerfile</emphasis> and builds a new image according to the
+      instructions.
+     </para>
+    </listitem>
+   </varlistentry>
+   <varlistentry>
+    <term>Container</term>
+    <listitem>
+     <remark>toms 2019-02-04: according to
+      https://docs.docker.com/glossary/?term=container a container consists
```
Comment (taroth21) [inline xml/docker_concept.xml]: I agree with Toms.

File: xml/docker_concept.xml [hunk 14]
```diff
@@ -0,0 +1,498 @@
...
+      <emphasis>container</emphasis> can be distinguished by a unique container
+      ID.
+     </para>
+    </listitem>
+   </varlistentry>
+   <varlistentry>
+    <term>Registry</term>
+    <listitem>
+     <para>
+      A <emphasis>registry</emphasis> is storage for already created images. It
```
Comment (taroth21) [inline xml/docker_concept.xml]: Shorten to: "A storage for already created images."

File: xml/docker_concept.xml [hunk 15]
```diff
@@ -0,0 +1,498 @@
...
+      typically contains several <emphasis>repositories</emphasis> There are two
+      types of registry:
+     </para>
+     <itemizedlist>
+      <listitem>
+       <formalpara>
+        <title>Public Registry:</title>
+        <para>
+        where everyone (usually registered) can download and
```
Comment (taroth21) [inline xml/docker_concept.xml]: What does "usually registered" refer to? Please clarify - is it about registering a user account at the registry, about registering a product? or even something else?). 

Also, similarly to varlistentries, I would shorten the explanation to:  "Everyone (usually registered [this needs to be defined, see above]  can download and use images.

File: xml/docker_concept.xml [hunk 16]
```diff
@@ -0,0 +1,498 @@
...
+        use images. A typical public registry is
+        <link xlink:href="https://hub.docker.com/">Docker Hub</link>.
+        </para>
+       </formalpara>
+      </listitem>
+      <listitem>
+       <formalpara>
+        <title>Private Registry:</title>
+        <para>
+        these are accessible for particular users or from a
```
Comment (taroth21) [inline xml/docker_concept.xml]: Rephrase to: "Only (?) accessible for particular users or users from a particular private network."

File: xml/docker_concept.xml [hunk 17]
```diff
@@ -0,0 +1,498 @@
...
+      </listitem>
+     </itemizedlist>
+    </listitem>
+   </varlistentry>
+   <varlistentry>
+    <term>Repository</term>
+    <listitem>
+     <para>
+      A <emphasis>repository</emphasis> is storage in a
+      <emphasis>registry</emphasis> that stores a different version of a
```
Comment (taroth21) [inline xml/docker_concept.xml]: Rephrase to: "A storage in the a ```<emphasis>registry</emphasis>``` that stores a different version of a particular image. You can pull images from a repository or push images to a repository."

File: xml/docker_concept.xml [hunk 18]
```diff
@@ -0,0 +1,498 @@
...
+      particular image. You can pull or push images from or to a repository.
+     </para>
+    </listitem>
+   </varlistentry>
+   <varlistentry xml:id="vle.docker.cgroup">
+    <term>Control group</term>
+    <listitem>
+     <para>
+      <emphasis>Control groups</emphasis>, also called
+      <literal>cgroups</literal>, is a Linux kernel feature that allows
```
Comment (taroth21) [inline xml/docker_concept.xml]: is->are (you are speaking about groups, so use a plural verb form).

File: xml/docker_concept.xml [hunk 19]
```diff
@@ -0,0 +1,498 @@
...
+      aggregating or partitioning tasks (processes) and all their children into
+      hierarchically organized groups to isolate resources.
```
Comment (taroth21) [inline xml/docker_concept.xml]: In this specific case, I would replace "to" with "in order to", otherwise it is hard to follow the sentence's meaning until the end.

File: xml/docker_concept.xml [hunk 20]
```diff
@@ -0,0 +1,498 @@
...
+     </para>
+    </listitem>
+   </varlistentry>
+   <varlistentry xml:id="vle.docker.namespace">
+    <term>Namespace</term>
+    <listitem>
+     <para>
+      &deng; uses <emphasis>namespaces</emphasis> for its containers that
```
Comment (taroth21) [inline xml/docker_concept.xml]: This sentence is a bit ambiguous (it's not clear to what 'that' refers: who isolates the resources?) Also: what are resources isolated *from*? from other resources reserved for other containers? 
To make it clearer, you could put it like that: "Used by &deng; for its containers to isolate resources reserved for particular containers (from???)."

File: xml/docker_concept.xml [hunk 21]
```diff
@@ -0,0 +1,498 @@
...
+ </sect1>
+ <sect1 xml:id="sec.docker.concept.advantages_deng">
+  <title>Advantages of &deng;</title>
+  <para>
+   &deng; provides you with the following advantages:
+  </para>
+  <itemizedlist>
+   <listitem>
+    <para>
+     Isolation of applications and operating systems through containers.
```
Comment (taroth21) [inline xml/docker_concept.xml]: As most of the list items are phrased verbally ("allows", "provides" etc.), phrase this sentence verbally, too: "Isolates applications and operating systems through containers."

File: xml/docker_concept.xml [hunk 22]
```diff
@@ -0,0 +1,498 @@
...
+    </para>
+   </listitem>
+   <listitem>
+    <para>
+     Near native performance, as &deng; manages allocation of resources in real
```
Comment (taroth21) [inline xml/docker_concept.xml]: To use a similar sentence structure like the rest of the list items, rephrase to: "Manage allocation of resources in real time, leading to a near-native performance."

File: xml/docker_concept.xml [hunk 23]
```diff
@@ -0,0 +1,498 @@
...
+     time.
+    </para>
+   </listitem>
+   <listitem>
+    <para>
+     Controls network interfaces and resources available inside containers
```
Comment (taroth21) [inline xml/docker_concept.xml]: This is ambiguous, too -> "Uses cgroups to control network interfaces and resources available inside containers."

File: xml/docker_concept.xml [hunk 24]
```diff
@@ -0,0 +1,498 @@
...
+    </para>
+   </listitem>
+  </itemizedlist>
+ </sect1>
+ <sect1 xml:id="sec.docker.concept.images-applications">
+  <title>&docker; Images of Applications</title>
+  <para>
+  &deng; is a technology that can help minimize resources used to run or
+  build applications. There are several types of applications that are
+  suitable to run inside a &docker; container like daemons, Web pages or
```
Comment (taroth21) [inline xml/docker_concept.xml]: insert comma after 'container'

File: xml/docker_concept.xml [hunk 25]
```diff
@@ -0,0 +1,498 @@
...
+ <sect2 xml:id="sec.docker.concept.images-applications.run-inside">
+  <title>Advantages of Running Applications Inside &docker; Containers</title>
+  <para>
+   Running an application inside a &docker; container has the
+   following advantages:
+  </para>
+  <itemizedlist>
+   <listitem>
+    <para>
+     You can minimize the runtime environment of the application as you can add
```
Comment (taroth21) [inline xml/docker_concept.xml]: "You can minimize the runtime environment of the application as you can add to the &docker; image of the application just the required processes and applications." -> "You can minimize the runtime environment of the application by adding just the required processes and applications to the &docker; image."

File: xml/docker_concept.xml [hunk 26]
```diff
@@ -0,0 +1,498 @@
...
+     to the &docker; image of the application just the required processes and
+     applications.
+    </para>
+   </listitem>
+   <listitem>
+    <para>
+     The image with your application is portable across machines also with
```
Comment (taroth21) [inline xml/docker_concept.xml]: "is portable" -> phrase verbally: "You can port the image with your application across machines with different Linux systems."

File: xml/docker_concept.xml [hunk 27]
```diff
@@ -0,0 +1,498 @@
...
+    </para>
+   </listitem>
+   <listitem>
+    <para>
+     You can share the image of your application by using a repository.
+    </para>
+   </listitem>
+   <listitem>
+    <para>
+     You can use different versions of required packages in the container than
```
Comment (taroth21) [inline xml/docker_concept.xml]: I'm not a native speaker but this does not sound correct to me. My proposal: "Within the container, you can use  package versions that are different from the ones used on the host system without causing dependency problems."

**Comment by zoopster** [comment]
I had thought documentation was to document, in this case, using SLE with docker. However this is an entire section on "why docker" vs "here's how to use SLE with docker". Wouldn't it be better to simply guide the user on where to find our images and how to build on top of them?

File: xml/docker_concept.xml [hunk 28]
```diff
@@ -0,0 +1,498 @@
...
+  </sect2>
+  <sect2 xml:id="sec.docker.concept.images-applications.build">
+   <title>Advantages of &deng; to Build Applications</title>
+   <para>
+    Using &deng; to build applications has the following advantages:
+   </para>
+   <itemizedlist>
+    <listitem>
+     <para>
+      You can prepare a complete building image.
```
Comment (taroth21) [inline xml/docker_concept.xml]: I have trouble understanding "complete building image":
- "complete build image"? 
- completely building image? (but that would not make any sense)

File: xml/docker_concept.xml [hunk 29]
```diff
@@ -0,0 +1,498 @@
...
+     </para>
+    </listitem>
+    <listitem>
+     <para>
+      Your build always runs in the same environment.
+     </para>
+    </listitem>
+    <listitem>
+     <para>
+      Developers can test their code in the same environment as used in
```
Comment (taroth21) [inline xml/docker_concept.xml]: "the same environment that is used in production"

File: xml/docker_concept.xml [hunk 30]
```diff
@@ -0,0 +1,498 @@
...
+   </itemizedlist>
+  </sect2>
+ </sect1>
+ <sect1 xml:id="sec.docker.concept.architecture_deng">
+  <title>&deng; Architecture</title>
+  <para>
+   &deng; uses a client/server architecture. You can use the <emphasis>CLI
+    client</emphasis> to communicate with the <emphasis>daemon</emphasis>. The
+   <emphasis>daemon</emphasis> then performs operations with containers and
+   manages images locally or in registry. The <emphasis>CLI client</emphasis>
```
Comment (taroth21) [inline xml/docker_concept.xml]: "in the registry"?

File: xml/docker_concept.xml [hunk 31]
```diff
@@ -0,0 +1,498 @@
...
+  </figure>
+ </sect1>
+
+ <sect1 xml:id="sec.docker.concept.definition_dreg">
+  <title>&dreg;</title>
+
+  <para>
+   The &dreg; is an open-source project created by Docker Inc. It
+   allows the storage and retrieval of &docker; images. By running a local
+   instance of the &dreg; it is possible to completely avoid the usage of
```
Comment (taroth21) [inline xml/docker_concept.xml]: insert comma after &degr;

File: xml/docker_concept.xml (hunk 8 above)
Comment (zoopster) [reply xml/docker_concept.xml]: a container image is unique. It is only used by a container engine. It is not a virtual machine. A "docker" image is a file, unique to docker and is made up of multiple layers each being read only except for the last layer.

File: xml/docker_concept.xml [hunk 32]
```diff
@@ -0,0 +1,498 @@
...
+  </para>
+
+  <para>
+   &dreg; is also used by &dhub;. However, &dhub;,
+   as seen from the user perspective, is made of the following parts at least:
+  </para>
+
+  <remark>toms 2019-02-05: Not sure if it is better to have separate sections
+  about terminology instead of one single one at the beginning.
+  Maybe you need to rephrase it to make it clear that the terms below refers
```
Comment (taroth21) [inline xml/docker_concept.xml]: ATM (while reading this PR online), I'm looking for language issues mostly. I will focus on the structure in a second review and then also look at this question in detail. I need to do the structure review in a printed version of the guide though, after those issues have been fixed.

File: xml/docker_concept.xml [hunk 33]
```diff
@@ -0,0 +1,498 @@
...
+  to the Docker Registry. Discuss with Tanja. :)
+  </remark>
+  <sect2 xml:id="sec.docker.concept.terminology_dreg">
+   <title>&dreg; Terminology</title>
+   <variablelist>
+    <varlistentry>
+     <term>
+      user interface (UI)
```
Comment (taroth21) [inline xml/docker_concept.xml]: Use title spelling for the term: User Interface (UI)

File: xml/docker_concept.xml (hunk 10 above)
Comment (zoopster) [reply xml/docker_concept.xml]: Sorry for the commentary...let's be clear, a docker image is a file for docker. It is not a template, not a virtual machine and is unique to docker. Actually I'd avoid "docker" altogether and replace with "container" as there are multiple engines available in SLE and CaaS Platform.

File: xml/docker_concept.xml (hunk 11 above)
Comment (zoopster) [reply xml/docker_concept.xml]: My previous comment applies here....containers are not virtual machines, images are not templates, they are simply files. A dockerfile is more like a manifest vs a recipe...it describes the building of the image, but doesn't have the parts to actually create the file.

File: xml/docker_concept.xml [hunk 34]
```diff
@@ -0,0 +1,498 @@
...
+     </term>
+     <listitem>
+      <para>
+       The <emphasis>user interface</emphasis> is the part that is accessed by
```
Comment (taroth21) [inline xml/docker_concept.xml]: Shorten to: "Accessed by users with their browser. Users can browse the contents of &dhub; either manually or by using a search feature." (avoid 'nice and intuitive').

The next sentence is also ambiguous:  "It also allows to create organizations made by different users." Do the organizations contain different sets of users? Can different users create organizations?

File: xml/docker_concept.xml [hunk 35]
```diff
@@ -0,0 +1,498 @@
...
+       It also allows to create organizations made by different users.
+      </para>
+      <para>
+       This component is closed-source.
+      </para>
+     </listitem>
+    </varlistentry>
+    <varlistentry>
+     <term>
+      authentication component
```
Comment (taroth21) [inline xml/docker_concept.xml]: use title spelling

File: xml/docker_concept.xml [hunk 36]
```diff
@@ -0,0 +1,498 @@
...
+     </term>
+     <listitem>
+      <para>
+       The <emphasis>authentication component</emphasis> is used to protect the images
```
Comment (taroth21) [inline xml/docker_concept.xml]: "Used to protect..."

File: xml/docker_concept.xml [hunk 37]
```diff
@@ -0,0 +1,498 @@
...
+       stored inside of &dhub;. It validates all push, pull and search requests.
+      </para>
+      <para>
+       This component is closed-source.
+      </para>
+     </listitem>
+    </varlistentry>
+    <varlistentry>
+     <term>
+      storage back-end
```
Comment (taroth21) [inline xml/docker_concept.xml]: use title spelling

File: xml/docker_concept.xml [hunk 38]
```diff
@@ -0,0 +1,498 @@
...
+     </term>
+     <listitem>
+      <para>
+       The <emphasis>storage back-end</emphasis> is where &docker; images are
```
Comment (taroth21) [inline xml/docker_concept.xml]: "Where &docker; images are ..."

File: xml/docker_concept.xml (hunk 14 above)
Comment (zoopster) [reply xml/docker_concept.xml]: " A registry is storage and content distribution for container images" <-- it does more than just store images. It's used to store, catalog, and retrieve images.

File: xml/docker_containers.xml [hunk 39]
```diff
@@ -37,18 +37,20 @@
...
- <para>
-  You may need to remove unused containers, you can achieve this by using:
- </para>
- <screen>&prompt.user;docker rm &lt;container name&gt;</screen>
-
+ <sect1 xml:id="Starting.Containers">
+  <title>Starting Containers</title>
+  <para>
+   The container typically exits if its main process finishes. For example, if
+   your container starts a particular application, as soon as you quit the
```
Comment (taroth21) [inline xml/docker_containers.xml]: re-order sentence parts: "For example, if your container starts a particular application, the container exits as soon as you quit the application." (easier to read)

File: xml/docker_containers.xml [hunk 40]
```diff
@@ -37,18 +37,20 @@
...
+   application, the container exits. You can start the container again by
+   running:
+  </para>
+  <screen>&prompt.user;docker start -ai &lt;container name&gt;</screen>
+  <para>
+   You may need to remove unused containers, you can achieve this by using:
```
Comment (taroth21) [inline xml/docker_containers.xml]: "To remove unused containers:" ```<screen>...</screen>```

File: xml/docker_containers.xml [hunk 41]
```diff
@@ -65,14 +67,13 @@
   <screen>&prompt.user;docker run -d --name sles sles12sp4 /bin/bash</screen>
 
   <para>
-   Then create a container that will link to the <emphasis>sles</emphasis>
-   container:
+   Then create a container that will link to the &slea; container:
```
Comment (taroth21) [inline xml/docker_containers.xml]: use &productname; entity instead of abbreviation

File: xml/docker_containers.xml [hunk 42]
```diff
@@ -65,14 +67,13 @@
...
   </para>
   <screen>&prompt.user;docker run --link sles:sles sles12sp4 /bin/bash</screen>
 
   <para>
-   The container that links to <emphasis>sles</emphasis> has defined
-   environment variables that enable connecting to the linked container.
+   The container that links to &slea; has defined environment variables that
```
Comment (taroth21) [inline xml/docker_containers.xml]: &productname;

File: xml/docker_installation.xml [hunk 43]
```diff
@@ -102,69 +102,90 @@
...
+     in consequence.
     </para>
    </step>
    <step>
     <para>
-     In case you will use Portus and an SSL secured registry, open the
-     <filename>/etc/sysconfig/docker</filename> file. Search for the parameter
-     <parameter>DOCKER_OPTS</parameter> and add <literal>--insecure-registry
-     <replaceable>ADDRESS_OF_YOUR_REGISTRY</replaceable></literal>.
+     If you will<remark>toms 2019-02-04: will -> want to?</remark> use
```
Comment (taroth21) [inline xml/docker_installation.xml]: yep, definitely: "want to"

File: xml/docker_installation.xml [hunk 44]
```diff
@@ -102,69 +102,90 @@
...
+     <step>
+      <para>Open the <filename>/etc/sysconfig/docker</filename> file. Search for
+       the parameter <parameter>DOCKER_OPTS</parameter> and add
+       <literal>--insecure-registry <replaceable>ADDRESS_OF_YOUR_REGISTRY</replaceable></literal>.
+      </para>
+     </step>
+     <step>
+      <para>
+       Add CA certificates to the directory
+       <filename>/etc/docker/certs.d/<replaceable>REGISTRY_ADDRESS</replaceable></filename>
```
Comment (taroth21) [inline xml/docker_installation.xml]: I would make this two separate steps:
Add the CA certificates to the directory...
Copy the CA certificates to your system...

File: xml/docker_registry.xml [hunk 45]
```diff
@@ -8,17 +8,18 @@
  xmlns="http://docbook.org/ns/docbook" version="5.0"
  xmlns:xi="http://www.w3.org/2001/XInclude"
  xmlns:xlink="http://www.w3.org/1999/xlink">
- <title>Storing Images</title>
+ <title>Configuring Image Storage</title>
  <para>
-  Prior to creating your own images, you should decide where you will store the
+  Prior to creating your own images, you should decide where you
+  will<remark>toms 2019-02-04: will -> want to?</remark> store the
```
Comment (taroth21) [inline xml/docker_registry.xml]: want to

**Comment by taroth21** [comment]
@m-rey : Hi Martin, many thanks! I added a lot of comments about details that I stumbled across when reading the PR online (none of the issues were caused by you, but we should fix them anyway during the revision of the guide). To look at the structure changes, I will generate a PDF from the latest version (hard to figure that out while reading online, let's discuss the structure in the printed version).

**Comment by taroth21** [comment]
> I had thought documentation was to document, in this case, using SLE with docker. However this is an entire section on "why docker" vs "here's how to use SLE with docker". Wouldn't it be better to simply guide the user on where to find our images and how to build on top of them?

@zoopster: I fully agree, I think we can get rid of a lot of stuff that has been in here for a long time, but is rather marketing-oriented and belongs somewhere else. However, our first step was to distinguish any kind of 'background information' from the actual instructions ('howtos') because both information types were mixed in many chapters. Now we have identified and separated those snippets, we can get rid of superfluous information and reduce the background information to the minimum (ideally condensed in one place). So anyone new to the topic can read the introduction to the concepts whereas the more experienced readers can skip this and go the instructions directly. 

Technical feedback and help in identifying what can be removed is appreciated. :)

**Comment by zoopster** [comment]
Ok good...point taken. I'll try to help contribute a bit more instead of drive-bys. Thanks for that detail, Tanja.

File: xml/docker_concept.xml (hunk 26 above)
Comment (zoopster) [reply xml/docker_concept.xml]: This isn't porting...it is consumable across container engines so it literally is "portable" not ported.

File: xml/docker_concept.xml (hunk 20 above)
Comment (m-rey) [reply xml/docker_concept.xml]: Fixed.
For more information about Docker namespaces, see [this article](https://success.docker.com/article/introduction-to-user-namespaces-in-docker-engine).

File: xml/docker_concept.xml (hunk 21 above)
Comment (m-rey) [reply xml/docker_concept.xml]: fixed

File: xml/docker_concept.xml (hunk 22 above)
Comment (m-rey) [reply xml/docker_concept.xml]: fixed

File: xml/docker_concept.xml (hunk 20 above)
Comment (m-rey) [reply xml/docker_concept.xml]: They are isolated from other containers or the host system.
https://success.docker.com/article/introduction-to-user-namespaces-in-docker-engine

File: xml/docker_concept.xml (hunk 34 above)
Comment (m-rey) [reply xml/docker_concept.xml]: Shortened.

Still not sure what is meant with "It also allows to create organizations made by different users."

File: xml/docker_concept.xml (hunk 15 above)
Comment (m-rey) [reply xml/docker_concept.xml]: There are official images, which are a curated images published by Docker.
There are also "Docker certified" images, which are images published directly by a commercial entity.
I don't know what the "usually registered" refers to.

File: xml/docker_concept.xml (hunk 11 above)
Comment (m-rey) [reply xml/docker_concept.xml]: @taroth21 changed.

File: xml/docker_concept.xml (hunk 10 above)
Comment (m-rey) [reply xml/docker_concept.xml]: @taroth21 Changed.

File: xml/docker_concept.xml (hunk 8 above)
Comment (m-rey) [reply xml/docker_concept.xml]: @taroth21 replaced "change" with "modification"

File: xml/docker_concept.xml [hunk 46]
```diff
@@ -70,32 +63,18 @@
...
-      paragraph:
-      "An image is a read-only template used to create a virtual machine on
-      the host server. Such an image is represented/build by using a
-      Dockerfile which acts as an "recipe". A Docker image is..."
-      </remark>
-      The changes are stored in a file called a
-      <emphasis>Dockerfile</emphasis>. For more details on Docker images, see
-      <link xlink:href="http://docs.docker.com/engine/reference/glossary#image">the
-       official Docker documentation</link>.
+      A self-contained piece of software. A <emphasis>&docker; image</emphasis>
```
Comment (zoopster) [inline xml/docker_concept.xml]: A docker image is simply a FILE. It is not self contained. It is not a virtual machine, either. The docker image relies on the container engine and the kernel of the host to run.
I think a better description is simply "a docker image is a file comprised of one or more layers used to execute code in a container"

**Comment by tomschr** [comment]
As Dima worked on the guide and merged it with the Container Guide, this PR can be closed now.
